testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -45,10 +46,28 @@
     private static final String TRINO_CATALOG = "iceberg";
     private static final String TEST_SCHEMA_NAME = "default";
 
-    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
-    public void testTrinoReadingSparkData()
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProviderClass = TestingIcebergStorageFormat.class, dataProvider = "unsupportedStorageFormats")
+    public void testTrinoWithUnsupportedFileFormat(TestingIcebergStorageFormat storageFormat)
     {
-        String baseTableName = "test_trino_reading_primitive_types";
+        String tableName = "test_trino_reading_primitive_types_" + storageFormat.name();
+        String trinoTableName = trinoTableName(tableName);
+        String sparkTableName = sparkTableName(tableName);
+
+        onSpark().executeQuery(format("CREATE TABLE %s (x bigint) USING ICEBERG TBLPROPERTIES ('write.format.default'='%s')", sparkTableName, storageFormat.name()));
+        onSpark().executeQuery(format("INSERT INTO %s VALUES (42)", sparkTableName));
+
+        assertQueryFailure(() -> onTrino().executeQuery("SELECT * FROM " + trinoTableName))
+                .hasMessageMatching("Query failed \\(#\\w+\\):\\Q File format not supported for Iceberg: " + storageFormat.name());
+        assertQueryFailure(() -> onTrino().executeQuery(format("INSERT INTO %s VALUES (42)", trinoTableName)))
+                .hasMessageMatching("Query failed \\(#\\w+\\):\\Q File format not supported for Iceberg: " + storageFormat.name());
+
+        onSpark().executeQuery("DROP TABLE " + sparkTableName);
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProviderClass = TestingIcebergStorageFormat.class, dataProvider = "storageFormats")
```

Comment:
Move the data provider to this class so that we don't need to add `dataProviderClass` everywhere. That also makes it more consistent with other tests like `TestHiveStorageFormats`.

Reply:
We are going to use this data provider in other test classes too.

Reply:
We won’t need this after we have Avro support. No need to make the code more complicated for a temporary situation.

Reply:
> We are going to use this data provider in other test classes too.

that was about test classes, not about Avro. I was explaining why i chose to use `dataProviderClass` here.